pennylane/plugins/tests/test_measurements.py

```diff
@@ -319,7 +319,7 @@ def circuit():
             qml.RX(theta, wires=[0])
             return qml.sample(qml.Hermitian(A_, wires=0))
 
-        res = circuit()
+        res = circuit().flatten()
```

Comment:
What issue did you run into here?

Reply:
The shared plugin test suite was returning a nested NumPy array here, which caused the `set(res)` call on line 327 to fail🤔 I'll dig into this more, I figured that the same would occur with vanilla `default.qubit`.

Reply:
Okay, I dug into it more.

The issue is that, in `BaseQNode`, we have the following logic:

```python
if res.return_type is ObservableReturnTypes.Sample:
    # Squeezing ensures that there is only one array of values returned
    # when only a single-mode sample is requested
    self.output_conversion = np.squeeze
```
This is so that a QNode with `return qml.sample(obs)` will return a 1D array of shape `(shots,)`.

In the `PassthruQNode`, however, we explicitly turn this off:

```python
# no output conversion
self.output_conversion = lambda x: x
```

The reason being that we don't know the ML interface the developer wants to use the `PassthruQNode` with, and don't want to accidentally cast output to NumPy arrays. So with the `PassthruQNode`, `return qml.sample(obs)` will be of shape `(1, shots)`.

Couple this with the fact that 1D NumPy arrays are hashable, but 2D arrays aren't, you run into the `set(res)` error here 🤔